### PR TITLE
Consolidate location managers

### PR DESCRIPTION
**Pull request by frederoni**
This PR breaks out a small part of #402 - reinvent the course tracking.
More specifically it combines `DefaultLocationManager` and `NavigationLocationManager` so that battery monitoring resides in core and we can reuse it when using `Simulated/ReplayLocationManager`.

@ericrwolfe @bsudekum 👀

### CODE REVIEW

File: MapboxCoreNavigation/NavigationLocationManager.swift [hunk 1]
```diff
@@ -10,6 +14,11 @@ open class NavigationLocationManager: CLLocationManager {
     
     var lastKnownLocation: CLLocation?
     
+    /**
+     Indicates whether the device is plugged in or not.
+     */
+    public var isPluggedIn: Bool = false
```
Comment (bsudekum) [inline MapboxCoreNavigation/NavigationLocationManager.swift]: Should this public and mutable? If it needs to be public can we make this a getter somehow?

Comment (frederoni) [inline MapboxCoreNavigation/NavigationLocationManager.swift]: fixed

File: MapboxCoreNavigation/NavigationLocationManager.swift [hunk 2]
```diff
@@ -27,5 +36,23 @@ open class NavigationLocationManager: CLLocationManager {
                 allowsBackgroundLocationUpdates = true
             }
         }
+        
+        #if os(iOS)
+            UIDevice.current.addObserver(self, forKeyPath: "batteryState", options: [.initial, .new], context: nil)
```
Comment (bsudekum) [inline MapboxCoreNavigation/NavigationLocationManager.swift]: Should this observer be removed on deinit?

Comment (frederoni) [inline MapboxCoreNavigation/NavigationLocationManager.swift]: Yep, that's already the case 😄

**Pull request merged**